code/modules/antagonists/mind_flayer/powers/flayer_passives.dm

```diff
@@ -0,0 +1,363 @@
+		if(istype(I, /obj/item/organ/internal/cyberimp))
+			var/obj/item/organ/internal/cyberimp/implant = I
+			implant.crit_fail = FALSE
+
+/datum/mindflayer_passive/eye_enhancement
+	name = "Enhanced Optical Sensitivity"
+	purchase_text = "Adjust our optical sensors to see better in the dark."
+	gain_text = "Focusing optics lens apeture."
+	upgrade_info = "Gain the ability to see prey through walls"
+	upgrade_text = "Increasing visible wavelength to infrared."
+	power_type = FLAYER_PURCHASABLE_POWER
+	max_level = 2
+	base_cost = 40
+	static_upgrade_increase = 20
+
+/datum/mindflayer_passive/eye_enhancement/on_apply()
+	..()
+	switch(level)
+		if(FLAYER_POWER_LEVEL_ONE)
+			ADD_TRAIT(owner, TRAIT_NIGHT_VISION, UNIQUE_TRAIT_SOURCE(src))
+		if(FLAYER_POWER_LEVEL_TWO)
+			ADD_TRAIT(owner, TRAIT_THERMAL_VISION, UNIQUE_TRAIT_SOURCE(src))
+	var/mob/living/carbon/human/to_enhance = owner //Gotta make sure it calls the right update_sight()
+	to_enhance.update_sight()
+
+/datum/mindflayer_passive/eye_enhancement/on_remove()
+	REMOVE_TRAIT(owner, TRAIT_NIGHT_VISION, UNIQUE_TRAIT_SOURCE(src))
+	REMOVE_TRAIT(owner, TRAIT_THERMAL_VISION, UNIQUE_TRAIT_SOURCE(src))
+	var/mob/living/carbon/human/to_enhance = owner
+	to_enhance.update_sight()
+
+/datum/mindflayer_passive/drain_speed
+	name = "Swarm Absorption Efficiency"
+	purchase_text = "Adds a multiplier to the amount of swarms you drain per second."
+	gain_text = "Our mental siphons grow stronger."
+	upgrade_text = "Energy transfer rate increased by 100%"
+	upgrade_info = "Further increase the rate of swarm siphoning."
+	power_type = FLAYER_PURCHASABLE_POWER
+	max_level = 3
+	base_cost = 50
+	static_upgrade_increase = 50
+
+/datum/mindflayer_passive/drain_speed/on_apply()
+	..()
+	flayer.drain_multiplier += 0.5
+
+/datum/mindflayer_passive/drain_speed/on_remove()
+	flayer.drain_multiplier = initial(flayer.drain_multiplier)
+
+/datum/mindflayer_passive/improved_joints
+	name = "Reinforced Joints"
+	purchase_text = "Prevents your limbs from falling off due to damage."
+	gain_text = "Artificial skeletal structure reinforced."
+	max_level = 1
+	power_type = FLAYER_PURCHASABLE_POWER
+	base_cost = 50
+
+/datum/mindflayer_passive/improved_joints/on_apply()
+	..()
+	ADD_TRAIT(owner, TRAIT_IPC_JOINTS_SEALED, UNIQUE_TRAIT_SOURCE(src))
+
+/datum/mindflayer_passive/improved_joints/on_remove()
+	REMOVE_TRAIT(owner, TRAIT_IPC_JOINTS_SEALED, UNIQUE_TRAIT_SOURCE(src))
+
+/datum/mindflayer_passive/telescopic_eyes
+	name = "Telescopic Eyes"
+	purchase_text = "Allows you to expand your sight range, as if you were using a scope."
+	gain_text = "Precise optics control engaged."
+	max_level = 1
+	power_type = FLAYER_PURCHASABLE_POWER
+	base_cost = 40
+	var/obj/item/organ/internal/eyes/optical_sensor/user_eyes
+
+/datum/mindflayer_passive/telescopic_eyes/on_apply()
+	. = ..()
+	user_eyes = owner.get_int_organ(/obj/item/organ/internal/eyes)
+	if(!user_eyes)
+		return // TODO, add a refund proc?
```

Comment:
TODO here

Reply:
Wasn't sure yet if this was gonna be in the full release, still not really sure